src/pdf/index.ts

```diff
@@ -17,10 +17,16 @@ export type TableOfContents = {
 
 let browser: Browser;
 
+puppeteer
+  .launch({ args: ['--disable-dev-shm-usage'] })
+  .then(inst => (browser = inst))
+  .catch(e => {
+    logger.logException('Failed to start browser puppeteer', e);
+  });
+
 export async function generatePdfFromHtml(html: string) {
-  if (!browser) {
-    browser = await puppeteer.launch({ args: ['--disable-dev-shm-usage'] });
-  }
+  // TODO: create browser lazily while keeping track of it
```

Comment:
I don't understand here what was the issue with the old way of checking if not created and creating the browser. Why you moved it outside and left this TODO.

Reply:
`generatePdfFromHtml` may get called concurrently while the browser launch promise is still pending, which results in ending up with multiple browser instances.

Reply:
Aha OK I get it now good to know. Great job! 💯